esp-hal/src/lib.rs

```diff
@@ -745,12 +745,42 @@ use crate::{
     peripherals::Peripherals,
 };
 
+/// Watchdog status.
+#[derive(Default, PartialEq)]
+pub enum WatchdogStatus {
+    /// Enables a watchdog timer with the specified timeout.
+    Enabled(fugit::MicrosDurationU64),
+    /// Disables the watchdog timer.
+    #[default]
+    Disabled,
+}
+
+/// Watchdogs configuration.
+#[non_exhaustive]
+#[derive(Default)]
+pub struct WatchdogConfig {
+    #[cfg(not(any(esp32, esp32s2)))]
+    /// Enable the super watchdog timer, which is slightly less than one second.
+    pub swd: bool,
+    /// Configures the reset watchdog timer.
+    pub rwdt: WatchdogStatus,
+    /// Configures the timg0 watchdog timer.
+    pub timg0: WatchdogStatus,
+    #[cfg(timg1)]
+    /// Configures the timg1 watchdog timer.
+    ///
+    /// By default, the bootloader does not enables this watchdog timer.
+    pub timg1: WatchdogStatus,
+}
```

Comment:
Not sure if this is the best place for these types to live, but I couldn't find any other better spot for them.

Reply:
> Maybe we should start a config module where we collect these

+1